python/ray/dashboard/dashboard.py

```diff
@@ -0,0 +1,325 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import aiohttp.web
+import argparse
+import datetime
+import json
+import logging
+import os
+import secrets
```

Comment:
Consider that the default behaviour of jupyter is to require a token because we have no idea whether users have exposed ports or not.

I'd be happy to include a --no-auth option (perhaps along with the --http-port option below) if necessary.

Replaced `secrets` with `SystemRandom` below.

Reply:
Two cases I'm thinking of:

Bookmarking for a dashboard, and refreshing a page after restarting a cluster. Having the URL be random would be impediment to both use cases.

Here I'm comparing to Spark UI which always has a fixed URL.

For security, I think it's sufficient to avoid binding to 0.0.0.0. Rather, by default you can only serve requests originating from the same machine (which ray port forward would also work for). That way you get network level security without any inconveniences.